handler.go

```diff
@@ -244,11 +252,42 @@ func (h *Handler) renderPage(w http.ResponseWriter, r *http.Request, path string
 		page.PreloadModules = template.HTML(h.manifest.GeneratePreloadModules(chunk.Src))
 	}
 
+	var tmplName string
+	if path == "/" {
+		tmplName = "index.html"
+	} else {
+		tmplName = path
+	}
+
 	// Find the template to use.
-	tmpl, ok := h.templates[path]
+	tmpl, ok := h.templates[tmplName]
+	if !ok {
+		// Try alternative names
+		alternativeNames := []string{
+			strings.TrimPrefix(tmplName, "/"),
+			strings.TrimPrefix(tmplName, "/") + ".html",
+			strings.TrimSuffix(strings.TrimPrefix(tmplName, "/"), ".html"),
+			tmplName + ".html",
+		}
+		for _, altName := range alternativeNames {
+			if t, found := h.templates[altName]; found {
+				tmpl = t
+				ok = true
+				break
+			}
+		}
+	}
+
 	if !ok {
-		// Use index.html by default
-		tmpl = h.templates["index.html"]
+		// check if custom templates were registered, perhaps a mistake was made in naming
+		if len(h.templates) > 1 {
+			keys := make([]string, 0, len(h.templates))
+			for k := range h.templates {
+				keys = append(keys, k)
+			}
+			log.Printf("Warning: template %q not found. Available templates: %s", tmplName, strings.Join(keys, ", "))
```

Comment:
Hmm... How do libraries log warnings nowadays? Do we use `log` or `slog.Warn(...)`, or invent a logger interface (which I don't like)?

Reply:
Thanks. I've been out of Go for a while and wasn't aware of the structured logging package. What do you think of something like this?

```go
	// Handle case when requested template is not found:
	// 1. If multiple templates exist, log a warning with the requested and available templates.
	// 2. Fall back to a default template.
	if !ok {
		if len(h.templates) > 1 {
			keys := make([]string, 0, len(h.templates))
			for k := range h.templates {
				keys = append(keys, k)
			}
			slog.Warn(
				"Template not found",
				"requestedTemplate", tmplName,
				"availableTemplates", strings.Join(keys, ", "),
			)
		}
		tmpl = h.templates[fallbackTemplateName]
	}
```